pkg/app/AppListingService.go

```diff
@@ -869,8 +862,9 @@ func (impl AppListingServiceImpl) FetchOtherEnvironment(ctx context.Context, app
 
 	ciArtifacts := make([]int, 0)
 	for _, env := range envs {
-		ciArtifacts = append(ciArtifacts, env.CiArtifactId)
-
+		if !slices.Contains(ciArtifacts, env.CiArtifactId) {
+			ciArtifacts = append(ciArtifacts, env.CiArtifactId)
```

Comment:
use map instead of slice, also create a function for filtering unique ids and use that at every place